pootle/apps/pootle_translationproject/models.py

```diff
@@ -277,6 +277,9 @@ def save(self, *args, **kwargs):
         else:
             self.abs_real_path = None
         super(TranslationProject, self).save(*args, **kwargs)
+        if self.directory.tp_id != self.pk:
+            self.directory.tp = self
+            self.directory.save()
```

Comment:
There is one for when you added the field, but if a deployment has current master (without this PR) and ran that migration and then added a new TP, then that new TP won't have the tp in the directories newly added. Thus another migration is needed, even if it is repeating the same thing.